header/interface.go

```diff
@@ -2,139 +2,30 @@ package header
 
 import (
 	"context"
-	"errors"
-	"fmt"
 
 	"github.com/ipfs/go-blockservice"
-	pubsub "github.com/libp2p/go-libp2p-pubsub"
-	tmbytes "github.com/tendermint/tendermint/libs/bytes"
-	core "github.com/tendermint/tendermint/types"
+	"github.com/tendermint/tendermint/types"
+
+	"github.com/celestiaorg/celestia-node/libs/header"
+	"github.com/celestiaorg/celestia-node/libs/header/p2p"
+	"github.com/celestiaorg/celestia-node/libs/header/sync"
 )
 
-// ConstructFn aliases a function that creates an ExtendedHeader.
+type Subscriber = header.Subscriber[*ExtendedHeader]
+type Subscription = header.Subscription[*ExtendedHeader]
+type Broadcaster = header.Broadcaster[*ExtendedHeader]
+type Exchange = header.Exchange[*ExtendedHeader]
+type Store = header.Store[*ExtendedHeader]
+type Getter = header.Getter[*ExtendedHeader]
+type Head = header.Head[*ExtendedHeader]
+type Syncer = sync.Syncer[*ExtendedHeader]
+type ExchangeServer = p2p.ExchangeServer[*ExtendedHeader]
```

Comment:
Question for the team. Do we want to keep those aliases or should we use the types directly?

Mainly, these aliases used in the node modules and are meant to prettify/hide generics(type parameters) from the code there. However, if you check it, the constructors(which don't alias her) for aliased components still require type parameters. So it seems like we are constructing one type but providing to the FX a completely different one, which is ambigious. So I would say we can safely remove these aliases. 

cc @renaynay and @distractedm1nd, who usuallly have opinions on such things

Reply:
Removed!